<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Erik Volz" />
  <meta name="dcterms.date" content="2020-01-31" />
  <title>How close are we to the ‘phylodynamic threshold’? A simulation study.</title>
  <style type="text/css">
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header>
<h1 class="title">How close are we to the ‘phylodynamic threshold’? A simulation study.</h1>
<p class="author">Erik Volz</p>
<p class="date">January 31, 2020</p>
</header>
<p>tl;dr: Almost there. Brush off your phylodynamic models.</p>
<p>Previous reports <a href="http://virological.org/t/phylodynamic-analysis-46-genomes-31-jan-2020/356?u=erik.volz">here</a> and <a href="http://virological.org/t/phylodynamic-analysis-of-ncov-2019-genomes-29-jan-2020/353/7?u=erik.volz">here</a> and <a href="http://virological.org/t/clock-and-tmrca-based-on-27-genomes/347?u=erik.volz">here</a> and <a href="http://virological.org/t/further-musings-on-the-tmrca/340?u=erik.volz">here</a> have shown that current sequences on GISAID and Genbank allow preliminary conclusions to be drawn about TMRCA and substitution rates. These studies have also found that it is more difficult to compute robust estimates of epidemic growth rates. This motivates the question: How much additional sequence data do we require and over what timeframe in order to get precise estimates of growth rates and by extension R0? We can investigate that question using simulations, which I report here.</p>
<p><strong><em>Key finding:</em></strong></p>
<p>If 50 additional sequences are provided over the first half of February, unbiased estimation of epidemic growth rates should be possible. Uncertainty in estimated substitution rates and TMRCA should also be drastically reduced.</p>
<h1 id="methods">Methods</h1>
<p>To facilitate computation over many simulations, I use maximum likelihood methods. All analyses are based on the <a href="https://nextstrain.org/ncov">Nextstrain</a> build from 30 January 2020 which includes a maximum likelihood tree of 42 sequences (cf acknowledgements table). Point estimates of TMRCA obtained from maximum likelihood methods appear consistent with recent reports using BEAST, but there is considerably more uncertainty in these estimates reflecting the importance of well-chosen &amp; informative priors with uninformative data.</p>
<p>I simulate coalescent trees conditioning on dates of sampling in the Nextstrain metadata and based on a SIR model calibrated to current epidemic conditions. Coalescent simulations are implemented in the <a href="https://github.com/emvolz-phylodynamics/phydynR">phydynR</a> package. Genealogical trees are generated under an infinite sites model using a substitution rate and TMRCA estimated using treedater and using the ML tree generated by Nextstrain. Time trees are then estimated from simulation outputs using <em>treedater</em>. The <a href="https://github.com/mrc-ide/skygrowth">skygrowth</a> package is then used to estimate a growth rate from estimated time trees.</p>
<p>Coalescent trees are based on a simple stochastic frequency-dependent SIR model where the rate of new infections is <br /><span class="math display"><em>β</em><em>S</em><em>I</em>/(<em>S</em> + <em>I</em> + <em>R</em>)</span><br /> and infections are removed at rate <br /><span class="math display"><em>γ</em><em>I</em></span><br />. The removal rate <br /><span class="math display"><em>γ</em> = 1/8.4</span><br /> is chosen so that the generation interval matches previous estimates and does not reflect incubation or infectious period: <a href="https://www.imperial.ac.uk/media/imperial-college/medicine/sph/ide/gida-fellowships/Imperial-2019-nCoV-transmissibility.pdf" class="uri">https://www.imperial.ac.uk/media/imperial-college/medicine/sph/ide/gida-fellowships/Imperial-2019-nCoV-transmissibility.pdf</a> The transmission rate <br /><span class="math display"><em>β</em></span><br /> is chosen to yeild an R0 value of 2.2.</p>
<p>Secondly, I consider the effect of adding 50 samples stratified over the first two weeks of February.</p>
<p>The steps involved in the simulation experiment are</p>
<ol type="1">
<li>An epidemic trajectory is simulated from the SIR stochastic differential equations</li>
<li>A coalescent tree is simulated conditioning on observed sample times and the SIR trajectory</li>
<li>A genealogical tree with branches in units of substitutions/site is generated under an infinite sites model. To reflect uncertainty in topology, this is re-estimated using neighbour joining.</li>
<li>A time tree is estimated with <em>treedater</em></li>
<li>The growth rate is estimated with <em>skygrowth</em></li>
</ol>
<h2 id="baseline-phylogenetic-analysis">Baseline phylogenetic analysis</h2>
<p><em>treedater</em> is invoked with</p>
<pre><code>td &lt;- dater( 
    tr # ML tree 
    , sts = sts # sample times
    , s = 29e3 # sequence length
    , estimateSampleTimes = est # ranges for tips with missing sample time
    , omega0 = c( 0.0005, 0.00075, .001) # initial guess of clock rate 
    , numStartConditions= 0
)</code></pre>
<p>One missing sample time was imputed. This yields the following maximum likelihood and time tree:</p>
<p><img src="/home/erikvolz/git/ncov/ranalyses/treedaterTimeTree-nextstrain30jan2020.png" /></p>
<p>I estimate the TMRCA to be 2019-11-28 (2019-02-15 – 2019-12-11). The confidence interval is based on parametric bootstrap which is illustrated here:</p>
<p><img src="/home/erikvolz/git/ncov/ranalyses/treedaterTMRCA-nextstrain30jan2020.png" /></p>
<p>I estimate the clock rate (substitutions/site/ year) to be 4.42e-04(1.31e-04 – 1.49e-03).</p>
<h2 id="simulation-results-based-on-current-samples">Simulation results based on current samples</h2>
<p>Simulated genealogies show a similar level of diversity as the real data and reveal difficulty of estimating time trees under these conditions: <a href="sim1compareTrees.png"></a> Left: A simulated coalescent tree, Middle: A genealogical tree (substitutions/site), Right: An estimated time tree</p>
<p>Simulated genealogical trees imply the following distribution of genetic diversity in simulations. These accord with current observations of number SNPs (red line):</p>
<p><a href="sim2_2020-02-01_snpDistribution.png"></a></p>
<p>Estimates of TMRCA show a slight tendency to under-estimate and the lower bound can not be robustly estimated with parametric bootstrap:</p>
<p><a href="sim2_2020-02-01_tmrcaEstimate.png"></a></p>
<p>Estimates of growth rate are highly variable; the distribution of point estimates overlap with the true value (red line) but there is a downwards bias:</p>
<p><a href="sim2_2020-02-01_grateDistribution.png"></a></p>
<h2 id="simulation-results-with-50-additional-samples-over-next-two-weeks">Simulation results with 50 additional samples over next two weeks</h2>
<p>Greater genetic diversity should be observed over next two weeks:</p>
<p><a href="sim2_2020-02-14_snpDistribution.png"></a></p>
<p>Estimates of TMRCA should become much more robust:</p>
<p><a href="sim2_2020-02-14_tmrcaEstimate.png"></a></p>
<p>Estimates of growth rate should fall into aligment with the true value, but we should still expect considerable uncertainty:</p>
<p><a href="sim2_2020-02-14_grateDistribution.png"></a></p>
<h1 id="caveates">Caveates</h1>
<ul>
<li>Epidemic dynamics are much more complex than portrayed by this SIR model. Notably, this simulation does not include variance in transmission rates, an incubation period, or geographic structure.</li>
<li>Infinite sites simulations do not account for rate variation.</li>
<li>Sampling is highly non-random. Cases from travellers are over-represented as are household and nosocomial transmissions. A real analysis would have to exclude redundant samples in sampled transmission chains. I have partially accounted for this by randomly downsampling sample dates to retain 30 samples. This is an approximation.</li>
<li>A <strong>well-designed</strong> Bayesian analysis should have superior performance, but this requires expert choice of informative priors.</li>
<li>Additional caveates mentioned by other posters regarding sequencing error also apply</li>
</ul>
<h1 id="code">Code</h1>
<p>All analysis and simulation code can be found at my fork of the nextstrain/ncov repository here:</p>
<p><a href="https://github.com/emvolz/ncov" class="uri">https://github.com/emvolz/ncov</a></p>
<h1 id="acknowledgements">Acknowledgements</h1>
<p>Developers and maintainers of Nextstrain deserve enormous credit for facilitating many steps of this analysis.</p>
<p>Data were shared by <a href="https://gisaid.org">GISAID</a>:</p>
<pre><code> * Shanghai Public Health Clinical Center &amp; School of Public Health, Fudan University, Shanghai, China
   - Wuhan-Hu-1/2019
 * National Institute for Viral Disease Control and Prevention, China CDC, Beijing, China
   - Wuhan/IVDC-HB-01/2019
   - Wuhan/IVDC-HB-04/2020
   - Wuhan/IVDC-HB-05/2019)
 * Institute of Pathogen Biology, Chinese Academy of Medical Sciences &amp; Peking Union Medical College, Beijing, China
   - Wuhan/IPBCAMS-WH-01/2019
   - Wuhan/IPBCAMS-WH-02/2019
   - Wuhan/IPBCAMS-WH-03/2019
   - Wuhan/IPBCAMS-WH-04/2019
 * Wuhan Institute of Virology, Chinese Academy of Sciences, Wuhan, China
   - Wuhan/WIV02/2019
   - Wuhan/WIV04/2019
   - Wuhan/WIV05/2019
   - Wuhan/WIV06/2019
   - Wuhan/WIV07/2019
 * Department of Microbiology, Zhejiang Provincial Center for Disease Control and Prevention, Hangzhou, China
   - Zhejiang/WZ-01/2020
   - Zhejiang/WZ-02/2020
 * Guangdong Provincial Center for Diseases Control and Prevention
   - Guangdong/20SF001/2020
   - Guangdong/20SF012/2020
   - Guangdong/20SF013/2020
   - Guangdong/20SF014/2020
   - Guangdong/20SF025/2020
   - Guangdong/20SF028/2020
   - Guangdong/20SF040/2020
   - Guangdong/20SF174/2020
   - Guangdong/20SF206/2020
   - Guangdong/20SF207/2020
   - Foshan/20SF207/2020
   - Foshan/20SF210/2020
   - Foshan/20SF211/2020
 * Department of Medical Sciences, National Institute of Health, Nonthaburi, Thailand
   - Nonthaburi/61/2020
   - Nonthaburi/74/2020
 * Division of Viral Diseases, Centers for Disease Control and Prevention, USA
   - USA-WA1/2020
   - USA/AZ1/2020
   - USA/IL1/2020
   - USA/CA1/2020
   - USA/CA2/2020
 * Centers for Disease Control, R.O.C., Taipei, Taiwan
   - Taiwan/2/2020
 * Institut Pasteur, Paris, France
   - France/IDF0372/2020
   - France/IDF0373/2020</code></pre>
</body>
</html>
